docs/best-practices.md

```diff
@@ -0,0 +1,82 @@
+- Return Timeout even with a non-zero number of events, if they are available
+- Return EOF (see [`sdk.ErrEOF`](https://github.com/falcosecurity/plugin-sdk-go/blob/0b4b6dc215141116c53398f3232aac98e49cdb80/pkg/sdk/sdk.go#L25) in Go or [`SS_PLUGIN_EOF`](https://github.com/falcosecurity/libs/blob/033c4b9f28e58e20a5822bd8a7419beea098af91/userspace/libscap/plugin_info.h#L77) in C/C++) when you are certain that no more events will be produced
+- Return EOF with a non-zero number of events, but ensuring that every following invokation of `plugin_next_batch()` returns EOF and zero events
+
+
+## Fields Naming Conventions
+
+Plugins can extend Falco by supporting new fields to be extracted event data. At runtime, those fields co-exist with the other fields natively supported in libsinsp, such as the `evt.*` or the `container.*` classes. In this perspective, it's really important for plugin developers to follow some naming conventions for the newly introduced fields.
+
+### Don't
+
+- Introduce fields with the same name or the same class (e.g. `evt.*`, `container.*`, `ka.*`) as the ones already existing in libsinsp
+- Use field names containing arbitrary characters
+- Assign unnecessarely long names to your fields
+
+### Do
+
+- Make sure your fields have an unique-ish class prefix that represents the use case of your plugin (e.g. `ct.*` for AWS Cloudtrail). This will help avoid ambiguities for rulesets using your fields in rule conditions
+- Use field names matching the following safe regular expression, that is compliant with the grammar of the rule parser: `[a-z]+[a-z0-9_]*(\.[a-z]+[a-z0-9_]*)+`
+- Use the `.` divisor character to give better meaning to your fields and divide them into categories (e.g. `ct.user.accountid` refers to the `user` category of AWS Cloudtrail events)
+
+
+## Live Events vs. Offline Events
+
+Once you get started developing a source plugin, you may come to the conclusion that there is absolutely no technical constraint on the events returned with `plugin_next_batch()`, since the plugin dictates both the data format of the events and how frequently they are produced. Some many degrees of freedom can be confusing and can make developer wonder to which extent the behavior of Falco should be changed.
+
+An important topic is the difference between producing events from a *live* or *offline* source. Those two terms are meant to distinguish events produced in real-time from the ones generated by parsing a static and reproducible source (such as a log file). In other words, live events are not reproducible because they are tied to real-time observations. To clear some of these doubts, be mindful of the following points:
```

Comment:
May we want to give the readers a simpler definition of *live* events too?

I would say something like:

live events do not include events that happened before the plugin startup.

And give also some examples. For instance, assuming one has a log file:
- `tail -f` would output live events only
- `cat` would output all events (ie. offline mode)